### PR TITLE
Fix release 18 again

### PR DESCRIPTION
**Pull request by systay**
## Description

This PR is on top of other planner bug fix PR that introduced a regression in a backport to v18.

## Related Issue(s)

- https://github.com/vitessio/vitess/pull/16962

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by arthurschreiber — APPROVED** [review]

File: go/vt/vtgate/planbuilder/operators/SQL_builder.go [hunk 1]
```diff
@@ -566,21 +566,16 @@ func buildProjection(op *Projection, qb *queryBuilder) error {
 func buildApplyJoin(op *ApplyJoin, qb *queryBuilder) error {
 	predicates := slice.Map(op.JoinPredicates, func(jc JoinColumn) sqlparser.Expr {
 		// since we are adding these join predicates, we need to mark to broken up version (RHSExpr) of it as done
-		qb.ctx.SkipPredicates[jc.RHSExpr] = nil
-
+		_ = qb.ctx.SkipJoinPredicates(jc.Original.Expr)
 		return jc.Original.Expr
 	})
+
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/operators/SQL_builder.go:572]: the error returned is not checked.
Should this be handled?

**Review by harshit-gangal — APPROVED** [review]

File: go/vt/vtgate/planbuilder/operators/SQL_builder.go (hunk 1 above)
Comment (arthurschreiber) [reply go/vt/vtgate/planbuilder/operators/SQL_builder.go:572]: I think we talked about this with @systay, and handling the error leads to some changes in behaviour and various tests failures. 🤔

Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/operators/SQL_builder.go:572]: It was for Foreign keys but that should not happen anymore.

Comment (systay) [reply go/vt/vtgate/planbuilder/operators/SQL_builder.go]: I've added the error checking. let's see what CI thinks 🤞

**Pull request merged**